### PR TITLE
[11.0] mig partner contact in several companies

### PR DESCRIPTION
**Pull request by njeudy**
- Migrate partner_contact_in_several_companies 
- Migrate partner_contact_personal_information_page

In the same PR because dependency

### CODE REVIEW

**Comment by njeudy** [comment]
I think a need to squash :)

But don't know how ?

**Comment by njeudy** [comment]
@pedrobaeza @gurneyalex don't understand runbot and codecov red bullet ... some help ?

**Comment by etobella** [comment]
The problem with the runbot is that your code raises a WARNING:
`2017-10-24 20:44:42,319 151 WARNING openerp_test odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view res.partner.kanban.contact (): use the hasclass(*classes) function to filter elements by their classes`
Codecov error will be solved when some PR are accepted (no initial report can be found). It should not be a major problem.

**Comment by njeudy** [comment]
Ok will change this :) thanks @etobella

**Comment by njeudy** [comment]
- Ok xpath expression changed
- And squash transbot commit

**Comment by njeudy** [comment]
@pedrobaeza all green :) ready for review ?

**Comment by pedrobaeza** [comment]
Well, I'm not directly interested in this module, so let's wait for other interested reviewers. Any way, I think if you rebase over current branch, you will get coverage, which is a good indicator also.

**Comment by njeudy** [comment]
@pedrobaeza ok no probleme will rebase. thanks

**Comment by njeudy** [comment]
@Yajo @pedrobaeza what should I do to pass codecov ? it is just 10.0 to 11.0 migration no improvement or new function ...

**Comment by yajo** [comment]
In this case codecov indicates that some merged code is not being tested. If you hit "Details > Diff", you'll see some red lines indicating that. You'd have to add some test for those lines to go over the target patch coverage (96.34%) and to avoid lowering the current project coverage ratio.

It would be a much appreciated effort, although I wouldn't require such thing IMHO.

**Comment by njeudy** [comment]
@Yajo I add some test but coverage checks was not launched .. ?

**Comment by njeudy** [comment]
@Yajo done ;)

**Review by yajo — APPROVED** [review]
Cool! Please squash all migration-related commits into one for the final merge 😊

**Comment by njeudy** [comment]
@Yajo like that ?

**Comment by yajo** [comment]
With https://github.com/OCA/partner-contact/pull/507/commits/867d552bdeb4712d34ba218f2c992650aaa6ce5b too

**Comment by njeudy** [comment]
@Yajo : did not understand why this commit is shown, in my branch all commit are squashed in one.
https://github.com/njeudy/partner-contact/commits/11.0-mig-partner_contact_in_several_companies ... don't know how to do.

**Comment by yajo** [comment]
@njeudy don't worry, let me show you. Look at the commits in the PR:

captura el 2017-11-15 a las 14 36 21

I highlighted all of the remaining translation commits.

The normal flow would be:

```
git checkout 11.0-mig-partner_contact_in_several_companies
git rebase --interactive 11.0
```

That should open the editor you have configured and show all the list of commits from your branch.

They should appear in this fashion:

```
pick <commit-sha> <commit-msg>
[...]
```

Now you should cut all those lines that I marked in red and move them to the bottom, just before your migration commit, and mark as `fixup` all of them except the first one. The end of the commits list should look something similar to this:

```
pick e112903 Create partner_contact_personal_information_page.
[...]
# Pick 1st translation commit
pick 4d5d348 OCA Transbot updated translations from Transifex 
# Fixup the rest of translation commits
fixup e639882 OCA Transbot updated translations from Transifex
# ...
fixup 2024a3e OCA Transbot updated translations from Transifex
# Your migration commit at the end:
pick 43e65d6 [MIG] V11 migration
```

Then save and exit. Git will start picking all code commits, then it will pick 1st translation commit and fixup (squash discarding commit message) all of other translation commits in that one, then it will pick your migration commit and will let a clean branch! There's chance that merge conflicts appear, in such case install `meld` and, when git rebase stops, do:

```
git mergetool
# now, using meld, fix conflicts on middle pane. Then:
git rebase --continue
```

If you even use that fixup system to remove the duplicate commits from your branch, that would be even more awesome, but this should be just fine with that.

**Comment by njeudy** [comment]
@Yajo thanks for your help, now it should be ok and have learn a lot if git rebase tricks :)

**Comment by yajo** [comment]
Yikes, I know you're gonna kill me, but just this little thing remaining in the rebase:

captura de pantalla de 2017-11-20 09-42-10

Remove the 1st of those 2 commits please. It's something that should have never happened. In fact, if you check it out you'll see it's empty: https://github.com/OCA/partner-contact/pull/507/commits/88e8d3c83c1707db573ecece0dbe6e709023b18a

To remove a commit in a `git rebase -i`, just delete or comment the commit line.

**Comment by njeudy** [comment]
@Yajo ok done :) thanks for the help and clean review.

**Comment by njeudy** [comment]
@Yajo @pedrobaeza ready to merge now ?

**Review by yajo — APPROVED** [review]

File: partner_contact_in_several_companies/__init__.py
Comment (simahawk) [inline partner_contact_in_several_companies/__init__.py]: aren't we dropping utf8 headers? Let's take a decision :smile: I have some pending PRs where I've already removed them and some to be updated... Just want to make sure we are doing the same thing for all the PRs/modules

File: partner_contact_in_several_companies/__manifest__.py
Comment (simahawk) [inline partner_contact_in_several_companies/__manifest__.py]: website should be https://github.com/OCA/partner-contact

File: partner_contact_in_several_companies/__manifest__.py
Comment (simahawk) [inline partner_contact_in_several_companies/__manifest__.py]: you can remove application, installable and auto_install

File: partner_contact_personal_information_page/README.rst
Comment (simahawk) [inline partner_contact_personal_information_page/README.rst:48]: update this part w/ new layout https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot
Comment (simahawk) [inline partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot]: aren't we excluding POT files?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot
Comment (pedrobaeza) [reply partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot]: POT files indeed should be removed

**Comment by simahawk** [comment]
A consideration about history: can't we squash little commits from the same authors? For instance:
image
Despite the fact that these commits should have be squashed in their PR we still have the chance to clean them up. 
We can group them w/ generic commits like "$name_of_the_module: misc fix/imp" and keep the sub-commits in the extended commit msg (as I did in #526).

File: partner_contact_in_several_companies/__init__.py
Comment (njeudy) [reply partner_contact_in_several_companies/__init__.py]: on all .py files ?

File: partner_contact_personal_information_page/README.rst
Comment (njeudy) [reply partner_contact_personal_information_page/README.rst:48]: Done

File: partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot
Comment (njeudy) [reply partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot]: done

File: partner_contact_in_several_companies/__manifest__.py
Comment (njeudy) [reply partner_contact_in_several_companies/__manifest__.py]: done

File: partner_contact_in_several_companies/__manifest__.py
Comment (njeudy) [reply partner_contact_in_several_companies/__manifest__.py]: done

**Comment by njeudy** [comment]
@simahawk : should I squash commit from same author ?

**Comment by simahawk** [comment]
@njeudy as long as they are not really important like "Introduced breaking change in API"... I'd say yes :)
But all the small changes I see there can be squashed by author for sure.

**Comment by njeudy** [comment]
@simahawk Ok done squash commit from same user in one

**Review by simahawk — APPROVED** [review]
LGTM. I'd squash commits like "make module uninstallable" and alike, just because they are useless in this context. But is not a blocker and we don't have guidelines for this ;)

**Comment by njeudy** [comment]
@pedrobaeza @simahawk @Yajo Ok to merge ?

**Review by yajo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
This has broken Travis. I have already fixed one issue, but it seems that there are more. This can't be installed. Please check https://travis-ci.org/OCA/partner-contact/jobs/317716741#L780

**Comment by pedrobaeza** [comment]
Please fix Travis after merging this.

**Comment by pedrobaeza** [comment]
Solved thanks to @drdran

**Pull request merged**
